fe/fe-core/src/main/java/org/apache/doris/catalog/external/HMSExternalTable.java

```diff
@@ -635,6 +636,30 @@ public void gsonPostProcess() throws IOException {
         super.gsonPostProcess();
         estimatedRowCount = -1;
     }
+
+    @Override
+    public List<Long> getChunkSizes() {
+        HiveMetaStoreCache.HivePartitionValues partitionValues = StatisticsUtil.getPartitionValuesForTable(this);
+        List<HiveMetaStoreCache.FileCacheValue> filesByPartitions
+                = StatisticsUtil.getFilesForPartitions(this, partitionValues, 0);
+        List<Long> result = Lists.newArrayList();
+        for (HiveMetaStoreCache.FileCacheValue files : filesByPartitions) {
+            for (HiveMetaStoreCache.HiveFileStatus file : files.getFiles()) {
+                result.add(file.getLength());
+            }
+        }
+        return result;
+    }
+
+    @Override
+    public long getDataSize(boolean singleReplica) {
+        List<Long> chunkSizes = getChunkSizes();
```

Comment:
Looks like it is a very heavy operation? It need to fetch all partitions and all files of a table, just for getting the total size of this table?
But in most case, the total size of table can be found in hive table's property?

Reply:
It is a heavy operation, as we discussed earlier, this brings a redundant fetching of all the files.
Usually  we can get total size of hive table in hms, but this call is not only to get the total size, but the size of each file. We use it to calculate the accurate sample ratio.

Reply:
Modified it to get total size from HMS first, only to fetch all files' size when HMS doesn't contain the total size value.